include/cudawrappers/cu.hpp

```diff
@@ -668,6 +668,249 @@ class DeviceMemory : public Wrapper<CUdeviceptr> {
+    _obj.srcZ = 0;
+    _obj.dstXInBytes = 0;
+    _obj.dstY = 0;
+    _obj.dstZ = 0;
+
+    _obj.WidthInBytes = size_x * element_size;
+    _obj.Height = size_y;
+    _obj.Depth = size_z;
+#endif
+  }
+};
+
+class GraphMemCopyToHostNodeParams : public Wrapper<CUDA_MEMCPY3D> {
+ public:
+  GraphMemCopyToHostNodeParams(void *dst, const CUdeviceptr &src,
+                               size_t element_size, size_t size_x,
+                               size_t size_y, size_t size_z, size_t pitch = 0) {
+#if defined(__HIP__)
+    memset(&_obj, 0, sizeof(hipMemcpy3DParms));
+    if (pitch == 0) {
+      pitch = size_x * element_size;
+    }
+    _obj.srcPtr =
+        make_hipPitchedPtr(const_cast<void *>(src), pitch, size_x, size_y);
+    _obj.dstPtr = make_hipPitchedPtr(const_cast<void *>(dst),
+                                     element_size * size_x, size_x, size_y);
+
+    _obj.extent = make_hipExtent(size_x * element_size, size_y, size_z);
+    _obj.kind = hipMemcpyDeviceToHost;
+#else
+    memset(&_obj, 0, sizeof(CUDA_MEMCPY3D));
+    _obj.srcMemoryType = CU_MEMORYTYPE_DEVICE;
+    _obj.dstMemoryType = CU_MEMORYTYPE_HOST;
+    _obj.srcDevice = src;
+    _obj.dstHost = dst;
+    _obj.srcXInBytes = 0;
+    _obj.srcY = 0;
+    _obj.srcZ = 0;
+    _obj.dstXInBytes = 0;
+    _obj.dstY = 0;
+    _obj.dstZ = 0;
+    _obj.WidthInBytes = size_x;
+    _obj.Height = size_y;
+    _obj.Depth = size_z;
+#endif
+  }
+};
+
+class Graph : public Wrapper<CUgraph> {
+ public:
+  explicit Graph(CUgraph &graph) : Wrapper(graph) {
+    checkCudaCall(cuCtxGetCurrent(&_context));
+  };
+
+  explicit Graph(unsigned int flags = CU_GRAPH_DEFAULT) {
+    checkCudaCall(cuGraphCreate(&_obj, flags));
+    manager = std::shared_ptr<CUgraph>(new CUgraph(_obj), [](CUgraph *ptr) {
+      checkCudaCall(cuGraphDestroy(*ptr));
+      delete ptr;
+    });
+    cuCtxGetCurrent(&_context);
+  }
+
+  void addKernelNode(GraphNode &node,
+                     const std::vector<CUgraphNode> &dependencies,
+                     GraphKernelNodeParams &params) {
+    checkCudaCall(cuGraphAddKernelNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_KERNEL_NODE_PARAMS *>(&params)));
+  }
+
+  void addHostNode(GraphNode &node,
+                   const std::vector<CUgraphNode> &dependencies,
+                   GraphHostNodeParams &params) {
+    checkCudaCall(cuGraphAddHostNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_HOST_NODE_PARAMS *>(&params)));
+  }
+
+  void addDevMemFreeNode(GraphNode &node,
+                         const std::vector<CUgraphNode> &dependencies,
+                         const CUdeviceptr &devPtr) {
+    checkCudaCall(cuGraphAddMemFreeNode(node.getNode(), _obj,
+                                        dependencies.data(),
+                                        dependencies.size(), devPtr));
+  }
+
+  void addDevMemAllocNode(GraphNode &node,
```

Comment:
Stick to the NVIDIA wording like with the other functions:
```suggestion
  void addMemAllocNode(GraphNode &node,
```